lib/app/admin/data-reset/test/remove-allItemsFromList-spec.js

```diff
@@ -0,0 +1,152 @@
+    async.eachSeries(sampleData.files(), function(file, eachMsgCb) {
+      mediatorRequestStub.withArgs(
+        'wfm:cloud:file:delete',
+        file.id,
+        {uid: file.id}
+      ).resolves(file.id);
+
+      remove.removeAllItemsFromList(mediatorMock, [file], 'file', function(err) {
+        assert(!err);
+        eachMsgCb(err);
+      });
+    }, function(err) {
+      assert(!err);
+      done();
+    });
+
+  });
+
+  it('Should remove all from group list', function(done) {
+    async.eachSeries(sampleData.group(), function(group, eachMsgCb) {
+      mediatorRequestStub.withArgs(
+        'wfm:cloud:group:delete',
+        group.id,
+        {uid: group.id}
+      ).resolves(group.id);
+
+      remove.removeAllItemsFromList(mediatorMock, [group], 'group', function(err) {
+        assert(!err);
+        eachMsgCb(err);
+      });
+    }, function(err) {
+      assert(!err);
+      done();
+    });
+
+  });
+
+  it('Should remove all from membership list', function(done) {
+    async.eachSeries(sampleData.membership(), function(membership, eachMsgCb) {
+      mediatorRequestStub.withArgs(
+        'wfm:cloud:membership:delete',
+        membership.id,
+        {uid: membership.id}
+      ).resolves(membership.id);
+
+      remove.removeAllItemsFromList(mediatorMock, [membership], 'membership', function(err) {
+        assert(!err);
+        eachMsgCb(err);
+      });
+    }, function(err) {
+      assert(!err);
+      done();
+    });
+
+  });
+
+  it('Should remove all from message list', function(done) {
+    async.eachSeries(sampleData.messages(), function(message, eachMsgCb) {
+      mediatorRequestStub.withArgs(
+        'wfm:cloud:messages:delete',
+        message.id,
+        {uid: message.id}
+      ).resolves(message.id);
+
+      remove.removeAllItemsFromList(mediatorMock, [message], 'messages', function(err) {
+        assert(!err);
+        eachMsgCb(err);
+      });
+    }, function(err) {
+      assert(!err);
+      done();
+    });
+
+  });
+
+  it('Should remove all from result list', function(done) {
+    async.eachSeries(sampleData.result(), function(result, eachMsgCb) {
+      mediatorRequestStub.withArgs(
+        'wfm:cloud:result:delete',
+        result.id,
+        {uid: result.id}
+      ).resolves(result.id);
+
+      remove.removeAllItemsFromList(mediatorMock, [result], 'result', function(err) {
+        assert(!err);
+        eachMsgCb(err);
+      });
+    }, function(err) {
+      assert(!err);
+      done();
+    });
+
+  });
+
+  it('Should remove all from workflows list', function(done) {
+    async.eachSeries(sampleData.workflows(), function(workflow, eachMsgCb) {
+      mediatorRequestStub.withArgs(
+        'wfm:cloud:workflows:delete',
+        workflow.id,
+        {uid: workflow.id}
+      ).resolves(workflow.id);
+
+      remove.removeAllItemsFromList(mediatorMock, [workflow], 'workflows', function(err) {
+        assert(!err);
+        eachMsgCb(err);
+      });
+    }, function(err) {
```

Comment:
So for things like this, it's easier and cleaner just to pass the reference to `done` instead of a function that does assertions and calls done().